### ISSUE TITLE
Semantics announcement not read on Desktop

### ISSUE THREAD

**Issue by yaakovschectman** [flutter/flutter#113059]
In the following code example, an announcement should be read, but none is heard.

Reproduce:
1. Run the below code example on Windows or MacOS with a screen reader running (NVDA, etc.),
2. Press the button.

Expected result:
- The announcement is read by the screen reader.

Actual result:
- No announcement is read.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/semantics.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Code Sample',
      home: Scaffold(
        body: IconButton(
          icon: const Icon(Icons.settings),
          tooltip: 'This is a tooltip',
          onPressed: () {
            SemanticsService.announce(
              'This is an announcement',
              TextDirection.ltr,
            );
          },
        ),
      ),
    );
  }
}
```

</details>

**Comment by yaakovschectman** [comment]
Currently, I do not think Flutter has any "announcement" support for Windows (please correct me if I have missed it). Additionally, we do not do anything when we receive the platform message for the announcement. The latter of these problems would involve wiring up a new built-in plugin like the `text_input_plugin` to handle accessibility channel messages. As for what we will do upon receiving these messages, I am still unsure.

**Comment by cppietime** [comment]
We probably need an approach like https://github.com/chromium/chromium/blob/main/ui/views/widget/root_view.cc#L76

**Comment by yaakovschectman** [comment]
First step will be to add a plugin that registers a method channel for `flutter/accessibility` in order to accept the announce method. Once that is up, my current plan is to attach a dummy AX node as a child of the root node with the alert text and issue an `ALERT` event on it.

**Comment by yaakovschectman** [comment]
Our approach had to be reverted, for now at least. I think there is some tricky COM-related requirement we aren't meeting now in our node implementation. Certain calls to its IAccessible from the NVDA python console result in `_ctypes.COMError: (-2147467262, 'No such interface supported', (None, None, None, 0, None))`, and Inspect shows the root node as having no children, so something isn't all the way there. The error message makes me suspect the COM.

**Comment by cbracken** [comment]
This was fixed in flutter/engine#37173
